src/classes/PermanentFileSystem.ts

```diff
@@ -72,6 +77,32 @@ export class PermanentFileSystem {
     ];
   }
 
+  public async getItemType(itemPath: string): Promise<number> {
+    if (isRootPath(itemPath)
+     || isArchiveCataloguePath(itemPath)
+     || isArchivePath(itemPath)
+     || isArchiveRootLevelFolderPath(itemPath)
+    ) {
+      return fs.constants.S_IFDIR;
+    }
+    const parentPath = path.dirname(itemPath);
+    const childName = path.basename(itemPath);
+    const parentFolder = await this.loadFolder(parentPath);
+    const targetIsRecord = parentFolder.records.some(
+      (record) => record.fileName === childName,
+    );
+    if (targetIsRecord) {
+      return fs.constants.S_IFREG;
+    }
+    const targetIsFolder = parentFolder.folders.some(
+      (folder) => folder.name === childName,
+    );
+    if (targetIsFolder) {
+      return fs.constants.S_IFDIR;
+    }
```

Comment:
This make sense!

Reply:
I just saw the second part of this!  I replied below in the main thread (copying the important concept here for posterity: basically the SDK doesn't understand the concept of a file path / doesn't think in terms of file system.)